From: Hannes Reinecke <hare@suse.de>
Date: Tue, 30 May 2017 11:49:12 +0200
Subject: [PATCH] loop: fix regression with op_is_write()
References: bsc#1034157
Patch-Mainline: no, solved in upstream via a rewrite

Commit 87374179c ("block: add a proper block layer data direction encoding")
introduced a regression with the loop driver, as the driver has been
using 'op_is_write()' as a short hand for decoding the actual operation.
With the mentioned commit op_is_write() checks the direction, not
the operation.

Signed-off-by: Hannes Reinecke <hare@suse.com>
Tested-by: Libor Pechacek <lpechacek@suse.com>
---
 drivers/block/loop.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/block/loop.c b/drivers/block/loop.c
index 48f6fa6..792dbf5 100644
--- a/drivers/block/loop.c
+++ b/drivers/block/loop.c
@@ -543,7 +543,7 @@ static int do_req_filebacked(struct loop_device *lo, struct request *rq)
 
 	pos = ((loff_t) blk_rq_pos(rq) << 9) + lo->lo_offset;
 
-	if (op_is_write(req_op(rq))) {
+	if (req_op(rq) != REQ_OP_READ) {
 		if (req_op(rq) == REQ_OP_FLUSH)
 			ret = lo_req_flush(lo, rq);
 		else if (req_op(rq) == REQ_OP_DISCARD)
-- 
1.8.5.6

